File: wms-lite-user-web/src/features/master/supplier/supplier.ts
/**
 * 공급업체 (Supplier) 도메인 관련 타입 정의
 * 백엔드 wms-lite-server (com.wms.wms_lite.domain.master.supplier) DTO 스펙 대응
 */

export interface SupplierResponse {
  id: number;
  code: string;
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
  address?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface SupplierSummaryResponse {
  id: number;
  code: string;
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
}

export interface SupplierSearchRequest {
  keyword?: string;
  page?: number;
  size?: number;
}

export interface SupplierCreateRequest {
  code: string;
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
  address?: string;
}

export interface SupplierUpdateRequest {
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
  address?: string;
}

File: wms-lite/wms-lite/src/main/java/com/wms/wms_lite/domain/master/supplier/entity/Supplier.java
package com.wms.wms_lite.domain.master.supplier.entity;

import com.wms.wms_lite.domain.master.supplier.enums.SupplierStatus;
import com.wms.wms_lite.global.entity.SoftDeleteEntity;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "suppliers")
public class Supplier extends SoftDeleteEntity {

    @Column(nullable = false, unique = true, length = 50)
    private String code;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(unique = true, length = 30)
    private String businessNo;

    private String ceoName;
    private String phone;
    private String email;
    private String address;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private SupplierStatus status = SupplierStatus.ACTIVE;
}
